### PR TITLE
Add Dataset output dict description

### PR DESCRIPTION
**Pull request by lucabergamini**
Add a description for the EgoDataset output in the `data_format.md`

- [X] Add EgoDataset description
- [x] Add AgentDataset description

### CODE REVIEW

File: data_format.md [hunk 1]
```diff
@@ -208,3 +208,96 @@ centroids = dt.agents[slice(10_000)]["centroid"]  # note this is the same as dt.
...
 
 **TL;DR**: when working with `zarr` you should always aim to minimise the number of accesses to the compressed data.
+
+## Dataset Abstraction Classes
+
+As shown above, working with the raw `zarr` dataset has its own perils. To that end, we provide two structures
+which form an additional abstraction layer over the raw `zarr` dataset. These two python classes allow to rasterise
+and get information about the past and future state of the AV or another agent. 
+
+**Note:** the following 2 classes inherit from Pytorch Dataset and as such are tied to work with it.
```
Comment (oliver-scheel) [inline data_format.md:218]: Id put just one **Note** and then two bullet points.

File: data_format.md [hunk 2]
```diff
@@ -208,3 +208,96 @@ centroids = dt.agents[slice(10_000)]["centroid"]  # note this is the same as dt.
...
+
+**Note:** the following 2 classes assume the world to be rasterised as BEV (Bird-Eye-View), this is a common choice for
```
Comment (oliver-scheel) [inline data_format.md:220]: which is a common choice

File: data_format.md [hunk 3]
```diff
@@ -208,3 +208,96 @@ centroids = dt.agents[slice(10_000)]["centroid"]  # note this is the same as dt.
...
+
+**Note:** the following 2 classes inherit from Pytorch Dataset and as such are tied to work with it.
+
+**Note:** the following 2 classes assume the world to be rasterised as BEV (Bird-Eye-View), this is a common choice for 
+CNN-based approaches. Still, this can be disabled by using `stub_debug` as `map_type`.
+
+
+### EgoDataset
+The `EgoDataset` retrieves information about the status of the AV in the current frame and the frames before it (if history is enabled).
+It can be iterated over the frame annotations to return a dict with the following fields:
```
Comment (oliver-scheel) [inline data_format.md:226]: This sentence doesn't make sense: 'It can be iterated over the frame annotations to return a dict with the following fields' - what do you mean?

File: data_format.md [hunk 4]
```diff
@@ -208,3 +208,96 @@ centroids = dt.agents[slice(10_000)]["centroid"]  # note this is the same as dt.
...
+
+```
+"image": the BEV raster as a multi-channel tensor;
```
Comment (oliver-scheel) [inline data_format.md:229]: Mhh I find this a bit hard to read in the code format ... but unfortunately also do not know what to do instead .. maybe a table?

File: data_format.md [hunk 5]
```diff
@@ -208,3 +208,96 @@ centroids = dt.agents[slice(10_000)]["centroid"]  # note this is the same as dt.
...
+"history_availabilities": same as target_availabilities but for the frames in the past;
+"world_to_image": a 3x3 matrix mapping from world to the image reference system; 
+"track_id": a scene-unique identifier id for the agent, or -1 for the AV;
+"timestamp": the timestamp of the current frame;
+"centroid": the centroid of the AV in the current frame in the world reference system. Unit is metres;
+"yaw": the angle of yaw of the AV in the current frame. Unit is radians;
+"extent": the extent of the AV (in XYZ) in the world reference system. Unit is metres;
+```
+
+An example of usage would be:
```
Comment (oliver-scheel) [inline data_format.md:244]: A sample usage would be:

File: data_format.md [hunk 6]
```diff
@@ -208,3 +208,96 @@ centroids = dt.agents[slice(10_000)]["centroid"]  # note this is the same as dt.
...
+    print(data["target_positions"])
+    print(data["history_positions"])
+```
+
+### AgentDataset
+The `AgentDataset` iterates over agents (i.e. every other dynamic entity in the scene) instead of the AV. Because the returned dict
+is exactly the same as the `EgoDataset`, the two classes are almost interchangeable. 
+
+However, one fundamental difference exists:
+The `AgentDataset` is seeded with an `agents_mask` which defines which agents should be iterated.
```
Comment (oliver-scheel) [inline data_format.md:270]: iterated over

File: data_format.md [hunk 7]
```diff
@@ -208,3 +208,96 @@ centroids = dt.agents[slice(10_000)]["centroid"]  # note this is the same as dt.
...
+### AgentDataset
+The `AgentDataset` iterates over agents (i.e. every other dynamic entity in the scene) instead of the AV. Because the returned dict
+is exactly the same as the `EgoDataset`, the two classes are almost interchangeable. 
+
+However, one fundamental difference exists:
+The `AgentDataset` is seeded with an `agents_mask` which defines which agents should be iterated. 
+This is used in multiple contexts:
+- to exclude unreliable agents during training (e.g. agents underneath a certain detection threshold);
+- to select a subset of agents (e.g. during evaluation for the competition)
+If the mask is not passed as an argument to the `AgentDataset`, a new one will be computed and **cached** based on the current value of `filter_agents_threshold`
```
Comment (oliver-scheel) [inline data_format.md:274]: Does not show a new line here in the resulting file.

Comment (oliver-scheel) [inline data_format.md:274]: . in the end of the sentence.

File: data_format.md (hunk 4 above)
Comment (lucabergamini) [reply data_format.md:230]: let me see if I can find a nice table for this :) great idea!

File: data_format.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (lucabergamini) [reply data_format.md]: mmm not really. `ChunkedDataset` managed data loading, but those two classes are tied to BEV. If you want to do something completely different, you shouldn't probably inherit from them right now. That's the base message here

File: data_format.md (hunk 3 above)
Comment (lucabergamini) [reply data_format.md:227]: that if you iterate over it you get a dict with those fields. I will rephrase it :)

File: data_format.md [hunk 8]
```diff
@@ -208,3 +208,98 @@ centroids = dt.agents[slice(10_000)]["centroid"]  # note this is the same as dt.
 we reduce the decompression numbers by **a factor of 10K**.
 
 **TL;DR**: when working with `zarr` you should always aim to minimise the number of accesses to the compressed data.
+
+## Dataset Abstraction Classes
+
+As shown above, working with the raw `zarr` dataset has its own perils. To that end, we provide two structures
+which form an additional abstraction layer over the raw `zarr` dataset. These two python classes allow to rasterise
```
Comment (oliver-scheel) [inline data_format.md:215]: Python

**Review by oliver-scheel — DISMISSED** [review]
Minor, LGTM!

**Review by oliver-scheel — APPROVED** [review]
LGTM
